### PR TITLE
Only include correct source files in cmake

### LINKED ISSUE #1050: Failing debian package build because of Abseil
I'm trying to build an updated cartographer_ros debian package (commit c0e713c404d3087cec3dbe07e35bd162ce802dc0 from  Wed Aug 29 12:21:47 2018)  for internal use as following (this worked before with version 1.0):

```
cd ros_cartographer
bloom-generate rosdebian --os-name ubuntu --os-version xenial --ros-distro kinetic
fakeroot debian/rules binary
```

The `cartographer_ros_msgs` and `cartographer` (commit a351a8e from Wed Sep 5 16:59:16 2018) debian packages build fine (and are installed).
Building with catkin works fine (even building install).

Errors:
```yaml
[ 25%] Building CXX object CMakeFiles/cartographer_ros.dir/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc.o
/usr/bin/c++   -DROSCONSOLE_BACKEND_LOG4CXX -DROS_PACKAGE_NAME=\"cartographer_ros\" -isystem /usr/include/lua5.2 -isystem /usr/include/pcl-1.7 -isystem /usr/include/eigen3 -isystem /opt/ros/kinetic/include -isystem /opt/ros/kinetic/share/xmlrpcpp/cmake/../../../include/xmlrpcpp -I/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu -I/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros -I/usr/src/gmock/gtest/include -isystem /usr/include/cairo -isystem /usr/include/glib-2.0 -isystem /usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem /usr/include/pixman-1 -isystem /usr/include/freetype2 -isystem /usr/include/libpng12 -isystem /opt/protobuf3/include -isystem /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2     -DEIGEN_USE_NEW_STDVECTOR -DEIGEN_YES_I_KNOW_SPARSE_MODULE_IS_NOT_STABLE_YET -pthread -std=c++11 -fPIC  -Wall -Wpedantic -Werror=format-security -Werror=missing-braces -Werror=reorder -Werror=return-type -Werror=switch -Werror=uninitialized -o CMakeFiles/cartographer_ros.dir/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc.o -c /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:111:1: error: ‘MutexImpl’ does not name a type
 MutexImpl::MutexImpl() {
 ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:116:1: error: ‘MutexImpl’ does not name a type
 MutexImpl::~MutexImpl() {
 ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:123:6: error: ‘MutexImpl’ has not been declared
 void MutexImpl::Lock() {
      ^
In file included from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/base/port.h:24:0,
                 from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/base/internal/low_level_alloc.h:51,
                 from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/mutex.h:65,
                 from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:19:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc: In function ‘void absl::synchronization_internal::Lock()’:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:124:38: error: ‘pthread_mutex_’ was not declared in this scope
   ABSL_RAW_CHECK(pthread_mutex_lock(&pthread_mutex_) == 0, "pthread error");
                                      ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:125:3: error: ‘locked_’ was not declared in this scope
   locked_ = true;
   ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc: At global scope:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:128:6: error: ‘MutexImpl’ has not been declared
 bool MutexImpl::TryLock() {
      ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc: In function ‘bool absl::synchronization_internal::TryLock()’:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:129:46: error: ‘pthread_mutex_’ was not declared in this scope
   bool locked = (0 == pthread_mutex_trylock(&pthread_mutex_));
                                              ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:130:15: error: ‘locked_’ was not declared in this scope
   if (locked) locked_ = true;
               ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc: At global scope:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:134:6: error: ‘MutexImpl’ has not been declared
 void MutexImpl::Unlock() {
      ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc: In function ‘void absl::synchronization_internal::Unlock()’:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:135:3: error: ‘locked_’ was not declared in this scope
   locked_ = false;
   ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:136:3: error: ‘released_’ was not declared in this scope
   released_.SignalAll();
   ^
In file included from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/base/port.h:24:0,
                 from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/base/internal/low_level_alloc.h:51,
                 from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/mutex.h:65,
                 from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:19:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:137:40: error: ‘pthread_mutex_’ was not declared in this scope
   ABSL_RAW_CHECK(pthread_mutex_unlock(&pthread_mutex_) == 0, "pthread error");
                                        ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc: At global scope:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:140:1: error: ‘CondVarImpl’ does not name a type
 CondVarImpl::CondVarImpl() {
 ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:145:1: error: ‘CondVarImpl’ does not name a type
 CondVarImpl::~CondVarImpl() {
 ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:149:6: error: ‘CondVarImpl’ has not been declared
 void CondVarImpl::Signal() {
      ^
In file included from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/base/port.h:24:0,
                 from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/base/internal/low_level_alloc.h:51,
                 from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/mutex.h:65,
                 from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:19:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc: In function ‘void absl::synchronization_internal::Signal()’:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:150:39: error: ‘pthread_cv_’ was not declared in this scope
   ABSL_RAW_CHECK(pthread_cond_signal(&pthread_cv_) == 0, "pthread error");
                                       ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc: At global scope:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:153:6: error: ‘CondVarImpl’ has not been declared
 void CondVarImpl::SignalAll() {
      ^
In file included from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/base/port.h:24:0,
                 from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/base/internal/low_level_alloc.h:51,
                 from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/mutex.h:65,
                 from /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:19:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc: In function ‘void absl::synchronization_internal::SignalAll()’:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:154:42: error: ‘pthread_cv_’ was not declared in this scope
   ABSL_RAW_CHECK(pthread_cond_broadcast(&pthread_cv_) == 0, "pthread error");
                                          ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc: At global scope:
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:157:6: error: ‘CondVarImpl’ has not been declared
 void CondVarImpl::Wait(MutexImpl* mu) {
      ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:157:24: error: variable or field ‘Wait’ declared void
 void CondVarImpl::Wait(MutexImpl* mu) {
                        ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:157:24: error: ‘MutexImpl’ was not declared in this scope
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:157:35: error: ‘mu’ was not declared in this scope
 void CondVarImpl::Wait(MutexImpl* mu) {
                                   ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:318:1: error: expected ‘}’ at end of input
 }  // namespace absl
 ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:318:1: error: expected ‘}’ at end of input
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:39:12: warning: ‘absl::Time absl::synchronization_internal::{anonymous}::DeadlineFromTimeout(absl::Duration)’ defined but not used [-Wunused-function]
 absl::Time DeadlineFromTimeout(absl::Duration timeout) {
            ^
/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc:46:12: warning: ‘absl::Time absl::synchronization_internal::{anonymous}::LimitedDeadline(absl::Time)’ defined but not used [-Wunused-function]
 absl::Time LimitedDeadline(absl::Time deadline) {
            ^
CMakeFiles/cartographer_ros.dir/build.make:1313: recipe for target 'CMakeFiles/cartographer_ros.dir/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc.o' failed
make[4]: *** [CMakeFiles/cartographer_ros.dir/abseil/src/abseil/absl/synchronization/internal/mutex_nonprod.cc.o] Error 1
make[4]: Leaving directory '/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu'
CMakeFiles/Makefile2:5220: recipe for target 'CMakeFiles/cartographer_ros.dir/all' failed
make[3]: *** [CMakeFiles/cartographer_ros.dir/all] Error 2
make[3]: Leaving directory '/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu'
Makefile:141: recipe for target 'all' failed
make[2]: *** [all] Error 2
make[2]: Leaving directory '/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/obj-x86_64-linux-gnu'
	cd /opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros
dh_auto_install: make -j1 install DESTDIR=/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros/debian/ros-kinetic-cartographer-ros AM_UPDATE_INFO_DIR=no returned exit code 2
debian/rules:57: recipe for target 'override_dh_auto_install' failed
make[1]: *** [override_dh_auto_install] Error 2
make[1]: Leaving directory '/opt/enway/cartographer_ws/src/cartographer_ros/cartographer_ros'
debian/rules:22: recipe for target 'binary' failed
make: *** [binary] Error 2
```

Shouldn't the `Abseil` build from the `cartographer` package be used for the ros package? (then I wouldn't run into these problems)

We got it to compile once, but had to manually install `https://github.com/google/benchmark` and also add the `${ABSEIL_INCLUDE_DIRS}`, add `ABSEIL` to the catkin package and then it worked with some weird combination of running `fakeroot debian/rules binary` then `fakeroot debian/rules` and then `fakeroot debian/rules binary` which points to some missing dependencies or compile race conditions...

### PR DESCRIPTION
**Pull request by redkite**
This fixes #1050. Tested standalone compilation and with the debian fakeroot tool. I had to build with a custom Protobuf3 instance though, so another build-check on a regular setup would be appreciated.

Before this change all *.cc files would be included. If subprojects
were run individually with the Debian package generator. This resulted in an
inclusion of temporarily checked out *.cc from the abseil include. This
change fixes the import behaviour and enables the creation of a valid
package.

### CODE REVIEW

File: cartographer_ros/CMakeLists.txt [hunk 1]
```diff
@@ -83,9 +83,9 @@ catkin_package(
   LIBRARIES ${PROJECT_NAME}
 )
 
-file(GLOB_RECURSE ALL_SRCS "*.cc" "*.h")
-file(GLOB_RECURSE ALL_TESTS "*_test.cc")
-file(GLOB_RECURSE ALL_EXECUTABLES "*_main.cc")
+file(GLOB_RECURSE ALL_SRCS "cartographer_ros/*.cc" "cartographer_ros/*.h")
+file(GLOB_RECURSE ALL_TESTS "cartographer_ros/*_test.cc")
+file(GLOB_RECURSE ALL_EXECUTABLES "cartographer_ros/*_main.cc")
```
Comment (MichaelGrupp) [inline cartographer_ros/CMakeLists.txt]: The `scripts` folder also has a `.cc` executable.

Comment (redkite) [reply cartographer_ros/CMakeLists.txt]: I don't have any `.cc` files in my scripts folder on the current master branch. Are some of these generated?

Comment (MichaelGrupp) [reply cartographer_ros/CMakeLists.txt]: True. I misread something :)

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user pifon2a. Merge queue now has a length of 1.

**Pull request merged**